deployment/data-streams/changeset/types/types.go

```
@@ -22,3 +24,25 @@ type (
 		OverrideRoot bool
 	}
 )
+
+type OwnershipFeature struct {
+	Transfer           bool // If true, MCMS takes ownership
+	DeployMCMS         bool
+	DeployMCMSConfig   commontypes.MCMSWithTimelockConfigV2
+	MCMSProposalConfig *proposalutils.TimelockConfig
+}
+
+func (f OwnershipFeature) AsSettings() OwnershipSettings {
+	if f.Transfer && f.MCMSProposalConfig == nil {
+		panic("MCMSConfig is required if Transfer is true")
```

I really don't like this `panic` here. We can return an error instead, which will make it possible for us to handle it ad maybe clean up.

If there is a reasonable chance that this struct can be in an inconsistent state, we should add a `Validate()` method, which we can call from other `Validate()` methods on struct which contain this.

---

addressed - pls review and lmk

---

Using a precondition function is great, thanks!